flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -396,6 +407,50 @@ private void stopTaskExecutorServices() throws Exception {
 		ExceptionUtils.tryRethrowException(exception);
 	}
 
+	private CompletableFuture<Void> stopTaskExecutorServices(
+			List<CompletableFuture<Void>> stoppingTaskFutures) throws Exception {
+		Exception exception = null;
+
+		try {
+			jobLeaderService.stop();
```

Comment:
This looks like duplicated code with the previous method `stopTaskExecutorServices`.
Can we split `stopTaskExecutorServices` into 2 methods and call them respectively in `stopTaskExecutorServices` and `onStop`? Then we do not need `stopTaskExecutorServices(stoppingTaskFutures)`.